keg_storage/backends/base.py

```diff
@@ -138,7 +138,8 @@ def download(
         path: str,
         file_obj: typing.IO,
         *,
-        progress_callback: typing.Optional[ProgressCallback] = None
+        progress_callback: typing.Optional[ProgressCallback] = None,
+        **kwargs,
```

Comment:
Accepting any `kwargs` here, because the possible options will depend on the storage backend. They are passed through to `open()` and applicable values are pulled out and used by the actual implementation.

I don't really like doing this, so let me know if you can think of a better way...